src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.cs

```diff
@@ -128,4 +128,117 @@ public async Task SourceMapsAllPathsInAttribute()
         await endpoint.RequestDelegate(httpContext);
         await VerifyResponseBodyAsync(httpContext, "Hello world!");
     }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task EmitsDiagnosticForUnsupportedAnonymousMethod(bool isAsync)
+    {
+        var source = isAsync
+            ? @"app.MapGet(""/hello"", async (int value) => await Task.FromResult(new { Delay = value }));"
+            : @"app.MapGet(""/hello"", (int value) => new { Delay = value });";
+        var (generatorRunResult, compilation) = await RunGeneratorAsync(source);
+
+        // Emits diagnostic but generates no source
+        var result = Assert.IsType<GeneratorRunResult>(generatorRunResult);
+        var diagnostic = Assert.Single(result.Diagnostics);
+        Assert.Equal(DiagnosticDescriptors.UnableToResolveAnonymousReturnType.Id, diagnostic.Id);
+        Assert.Empty(result.GeneratedSources);
```

Comment:
Should there be test(s) for emitting source for valid endpoints when invalid endpoints are also used? The diagnostic messages sort of imply that generation continues, and the messages could be suppressed technically.

Reply:
> Should there be test(s) for emitting source for valid endpoints when invalid endpoints are also used?

Good recommendation.